k2/python/k2/fsa.py

```diff
@@ -26,6 +26,57 @@ class Fsa(object):
     a vector of FSAs it is a tuple with three
     elements ``(num_fsas, None, None)``.  (Caution: it's possible
     for a vector of FSAs to have zero or one elements).
+
+    An instance of FSA has the following attributes:
+
+    - ``arcs``: You will NOT use it directly in Python. It is an instance
+                of ``_k2.RaggedArc`` with only one method ``values()`` which
+                returns a 2-D `torch.Tensor`` of dtype ``torch.int32`` with 4
+                columns. Its number of rows indicates the number of arcs in the
+                FSA. The first column represents the source states, second
+                column the destination states, third column the labels and the
+                fourth column is the score. Note that the score is actually
+                a float number but it is **reinterpreted** as an integer.
+
+    - ``score``: A 1-D ``torch.Tensor`` of dtype ``torch.float32``. It has
+                 as many entries as the number of arcs representing the score
+                 of every arc.
+
+    - ``labels``: A 1-D ``torch.Tensor`` of dtype ``torch.int32``. It has as
+                  many entries as the number of arcs representing the label of
+                  every arc.
+
+    - ``arc_labels_sorted``: A boolean. True if arcs of every state are sorted
```

Comment:
Can someone remind me why we sort arcs on labels first then dest_state in host code? I believe there was a reason that time. And seems now in k2 code, we just sort on labels?
https://github.com/k2-fsa/k2/blob/6acfe3d54140889a8456a4002cb393607c165c64/k2/csrc/fsa_algo.cu#L236-L250

Reply:
As far as I know, sorting is to make `intersect` run faster. `dest` state does not affect the intersection process, only the labels matter. The current implementation assumes both the two input FSAs are arc sorted.

Reply:
Seems that time we wanted to use this feature in `determinize`, but finally we didn't use it. https://github.com/k2-fsa/k2/pull/17#issuecomment-620587443
@danpovey Should we change the implementation in k2 for the `operator<` in Arc to just sort on labels?

https://github.com/k2-fsa/k2/blob/6acfe3d54140889a8456a4002cb393607c165c64/k2/csrc/fsa.h#L37-L42

Reply:
Just sorting on labels should be OK for now.  But it will make very little difference to the speed.